calabash-cucumber/lib/calabash-cucumber/keyboard_helpers.rb

```diff
@@ -188,12 +188,61 @@ def text_from_first_responder
         text
       end
 
+      # @!visibility private
+      # Returns the keyboard type as a symbol from the specified query
+      #
+      # UIKeyboardTypeDefault => :default
+      # UIKeyboardTypeASCIICapable => :ascii_capable
+      # UIKeyboardTypeNumbersAndPunctuation => :numbers_and_punctuation
+      # UIKeyboardTypeURL => :url
+      # UIKeyboardTypeNumberPad => :number_pad
+      # UIKeyboardTypePhonePad => :phone_pad
+      # UIKeyboardTypeNamePhonePad => :name_phone_pad
+      # UIKeyboardTypeEmailAddress => :email
+      # UIKeyboardTypeDecimalPad => :decimal
+      # UIKeyboardTypeTwitter => :twitter
+      # UIKeyboardTypeWebSearch => :web_search
+      #
+      # @raise [RuntimeError] if there is no visible keyboard
+      def keyboard_type(query)
+        if !keyboard_visible?
+          screenshot_and_raise "There must be a visible keyboard"
+        end
+
+        if query.nil? || query.empty?
+          # Default to first responder
+          query = "* isFirstResponder:1"
+        end
+        keyboard_type = _query_wrapper(query, :keyboardType).first
+        if keyboard_type.is_a?(Fixnum)
+          return KEYBOARD_TYPES[keyboard_type]
+        else
+          screenshot_and_raise "Invalid keyboard type"
```

Comment:
Should this raise?

In this case a screenshot would not give enough information to help the user or us understand what went wrong.

The way I can see this failing is:

1. keyboardType is nil - either there was no first responder (empty results) or for some reason the first element of the results is `nil`.
2. keyboardType is "*****" - first responder does not respond to `keyboardType`.
3. keyboardType is unknown to us - we don't know about it so it is not one of our KEYBOARD_TYPES
4. keyboardType is not a Fixnum ^ similar to above

Point 2 above will happen fairly often - I believe the Slack app, for example, is using a UI Text Input class that does not respond to `keyboardType`.  There is an example of this kind of input field in the CalSmokeApp feature/update-to-0.20.3.

If we do raise, I think we should try to provide as much information as possible (capture why we failed).

Maybe we debug log the exceptional cases and return `:unknown` if we are on the unhappy path?=

```
query_result = _query_wrapper(query, :keyboardType).first

keyboard_type = KEYBOARD_TYPES[query_result]

if !keyboard_type
  # Assuming we do not raise
  keyboard_type = :unknown
end

keyboard_type
```